File: crates/nil-payload/src/player.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::player::{PlayerId, PlayerOptions, PlayerStatus};
use nil_core::world::WorldId;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPlayerRequest {
  pub world: WorldId,
  pub id: PlayerId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPlayerCoordsRequest {
  pub world: WorldId,
  pub id: PlayerId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPlayersRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPlayerMaintenanceRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPlayerMilitaryRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPlayerReportsRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPlayerStatusRequest {
  pub world: WorldId,
  pub id: PlayerId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPlayerStorageCapacityRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPublicPlayerRequest {
  pub world: WorldId,
  pub id: PlayerId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPublicPlayersRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct PlayerExistsRequest {
  pub world: WorldId,
  pub id: PlayerId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct SetPlayerStatusRequest {
  pub world: WorldId,
  pub status: PlayerStatus,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct SpawnPlayerRequest {
  pub world: WorldId,
  pub options: PlayerOptions,
}

File: app/src/lib/request/player.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

export interface GetPlayerRequest {
  readonly world: WorldId;
  readonly id: PlayerId;
}

export interface GetPlayerCoordsRequest {
  readonly world: WorldId;
  readonly id: PlayerId;
}

export interface GetPlayersRequest {
  readonly world: WorldId;
}

export interface GetPlayerMaintenanceRequest {
  readonly world: WorldId;
}

export interface GetPlayerMilitaryRequest {
  readonly world: WorldId;
}

export interface GetPlayerReportsRequest {
  readonly world: WorldId;
}

export interface GetPlayerStatusRequest {
  readonly world: WorldId;
  readonly id: PlayerId;
}

export interface GetPlayerStorageCapacityRequest {
  readonly world: WorldId;
}

export interface GetPublicPlayerRequest {
  readonly world: WorldId;
  readonly id: PlayerId;
}

export interface GetPublicPlayersRequest {
  readonly world: WorldId;
}

export interface PlayerExistsRequest {
  readonly world: WorldId;
  readonly id: PlayerId;
}

export interface SetPlayerStatusRequest {
  readonly world: WorldId;
  readonly status: PlayerStatus;
}

export interface SpawnPlayerRequest {
  readonly world: WorldId;
  readonly options: PlayerOptions;
}
